SOD.Common/Patches/RewiredPlayerPatches.cs

```diff
@@ -0,0 +1,77 @@
+using System.Linq;
+using System.Reflection;
+using HarmonyLib;
+using Rewired;
+using SOD.Common.Shadows;
+using SOD.Common.Shadows.Implementations;
+using UnityEngine;
+
+namespace SOD.Common.Patches
+{
+    internal class RewiredPlayerPatches
+    {
+        [HarmonyPatch]
+        internal class RewiredPlayer_GetButtonDown
+        {
+            internal static int lastFrameCount = -1;
+
+            [HarmonyTargetMethod]
+            internal static MethodBase CalculateMethod()
+            {
+                return typeof(Rewired.InputManager_Base).Assembly.ExportedTypes
+                    .Single(t => t.FullName == "Rewired.Player")
+                    .GetMethods()
+                    .Where(m => m.Name == "GetButtonDown")
+                    .Single(m => m.GetParameters().Single().ParameterType == typeof(string));
```

Comment:
rework to FirstOrDefault() with descriptive error messages, (I would not use Single or SingleOrDefault) as they will still give sequence errors if there is more than 1 element

Reply:
N/A

Reply:
No longer needed with changes to patch